warehouse/email/__init__.py

```diff
@@ -56,8 +73,16 @@ def _send_email_to_user(request, user, msg, *, email=None, allow_unverified=Fals
     if email is None or not (email.verified or allow_unverified):
         return
 
+    # We should only store/display IP address of an 'email sent' event if the user
+    # who triggered the email event is the one who receives the email. Else display
+    # 'Redacted' to prevent user privacy concerns
+    user_email = request.db.query(Email).filter(Email.email == email.email).one()
+    ip_address = (
+        request.remote_addr if user_email.user_id == request.user.id else "Redacted"
+    )
```

Comment:
Why not:

```suggestion
    ip_address = (
        request.remote_addr if email == request.user.primary_email else "Redacted"
    )
```
?

Reply:
The user_email was queried in the off-chance that an email is sent to a non-primary email, such as when sending a `verify-email` email.